scripts/cat/enums/status.py

```diff
@@ -0,0 +1,157 @@
+from scripts.game_structure.extended_strenum import ExtendedStrEnum
+
+
+class StatusEnum(ExtendedStrEnum):
+    """
+    Roles that a cat may hold. Check ExtendedStrEnum for more things!
+    """
+    # These are in this order for food priority filtering
+    NEWBORN = "newborn"
+    KITTEN = "kitten"
+    ELDER = "elder"
+    WARRIORAPP = "apprentice"
+    WARRIOR = "warrior"
+    MEDIATORAPP = "mediator apprentice"
+    MEDIATOR = "mediator"
+    MEDCATAPP = "medicine cat apprentice"
+    MEDCAT = "medicine cat"
+    DEPUTY = "deputy"
+    LEADER = "leader"
+
+    NONE = "ERROR"
+    EXCLAN = "former Clancat"
+    EXILED = "exiled"
+    KITTYPET = "kittypet"
+    LONER = "loner"
+    ROGUE = "rogue"
+
+    def is_working_any(self):
+        """True if cat is a working role (warrior + app, medcat + app, mediator + app, deputy and leader)"""
+        return not (self.is_kit_any()
+                    or self.is_elder()) \
+            and self.is_clan_status()
+
+    def is_patrol_any(self):
+        """True if able to go on patrol (warrior + app, medcat + app, deputy and leader)"""
+        return (self.is_warrior_any()
+                or self.is_medcat_any()
+                or self.is_deputy_or_leader())
+
+    def is_patrol_app(self):
+        """True if apprentice who can go on patrol (warrior app and medcat app)"""
+        return self in [StatusEnum.WARRIORAPP, StatusEnum.MEDCATAPP]
+
+    def is_kit_any(self):
+        """True if cat is newborn or kitten."""
+        return self in [StatusEnum.NEWBORN, StatusEnum.KITTEN]
+
+    def is_app_any(self):
+        """True if cat is apprentice,
+        medicine cat apprentice or mediator apprentice."""
+        return self in [StatusEnum.WARRIORAPP,
+                        StatusEnum.MEDCATAPP,
+                        StatusEnum.MEDIATORAPP]
+
+    def is_mediator_any(self):
+        """True if cat is mediator or mediator apprentice."""
+        return self in [StatusEnum.MEDIATOR, StatusEnum.MEDIATORAPP]
+
+    def is_medcat_any(self):
+        """True if cat is medicine cat or medicine cat apprentice."""
+        return self in [StatusEnum.MEDCAT, StatusEnum.MEDCATAPP]
+
+    def is_warrior_any(self):
+        """True if cat is warrior or apprentice."""
+        return self in [StatusEnum.WARRIOR, StatusEnum.WARRIORAPP]
+
+    def is_deputy_or_leader(self):
+        """True if cat is deputy or leader."""
+        return self in [StatusEnum.DEPUTY, StatusEnum.LEADER]
+
+    def is_warrior_medcat_or_mediator(self):  # TODO grr.
+        """True if cat is warrior, medcat or mediator"""
+        return self in [StatusEnum.WARRIOR, StatusEnum.MEDCAT, StatusEnum.MEDIATOR]
+
+    def is_normal_adult(self):
```

Comment:
What does `normal` mean? I don't think that's the most fitting word.

Reply:
I was told to use this phrasing because it matches the patrol phrasing. I'm not the biggest fan of it either - `is_deputy_leader_or_warrior` is "worse" but also clearer, so will swap to that